gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Channelizer.java

```diff
@@ -149,6 +155,18 @@ protected void initChannel(final SocketChannel socketChannel) {
             configure(pipeline);
             pipeline.addLast(PIPELINE_GREMLIN_HANDLER, new GremlinResponseHandler(pending));
         }
+
+        @Override
+        public void connected() {
+            if (supportsSsl()) {
+                try {
+                    // Block until the handshake is complete either successfully or with an error.
+                    sslHandler.handshakeFuture().sync();
+                } catch (Exception ex) {
+                    throw new ConnectionException(connection.getUri(), HANDSHAKE_ERROR, ex);
```

Comment:
This might leak the channel. Can you confirm if something actually closes the channel if an exception is thrown here?

Reply:
When there is a handshake failure, the SslHandler closes the channel via call chain: SslUtils.handleHandshakeFailure-->ChannelHandlerContext.close().